webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/DeploymentService.scala

```diff
@@ -132,56 +255,109 @@ trait PromiseGenerationService extends Loggable {
 
     import HooksImplicits._
 
+    /*
+     * The computation of dynamic group is a workaround inconsistencies after importing definition
+     * from LDAP, see: https://issues.rudder.io/issues/14758
+     * But that computation may lead to a huge delay in generation (tens of minutes). We want to be
+     * able to unset their computation in some case through an environement variable.
+     *
+     * To disable it, set environment variable "rudder.generation.computeDynGroup" to "disabled"
+     */
+    val computeDynGroupsEnabled = getComputeDynGroups().getOrElse(true)
+
+    val maxParallelism = {
+      // We want to limit the number of parallel execution and threads to the number of core/2 (minimum 1) by default.
+      // This is taken from the system environment variable because we really want to be able to change it at runtime.
+      def threadForProc(mult: Double): Int = {
+        Math.max(1, (Runtime.getRuntime.availableProcessors * mult).ceil.toInt)
+      }
+      val t = try {
+        getMaxParallelism().getOrElse("x0.5") match {
+          case s if s.charAt(0) == 'x' => threadForProc(s.substring(1).toDouble)
+          case other => other.toInt
+        }
+      } catch {
+        case ex: IllegalArgumentException => threadForProc(1)
+      }
+      if(t < 1) {
+        PolicyLogger.warn(s"You can't set 'rudder_generation_max_parallelism' so that there is less than 1 thread for generation")
+        1
+      } else {
+        t
+      }
+    }
+    val jsTimeout = {
+      // by default 5s but can be overrided
+      val t = getJsTimeout().getOrElse(5)
+      FiniteDuration(try {
+        Math.max(1, t.toLong) // must be a positive number
+      } catch {
+        case ex: NumberFormatException =>
+          PolicyLogger.error(s"Impossible to user property '${t}' for js engine timeout: not a number")
+          5L // default
+      }, TimeUnit.SECONDS)
+    }
+
+    PolicyLogger.debug(s"Policy generation parallelism set to: ${maxParallelism} (change with REST API settings parameter 'rudder_generation_max_parallelism')")
+    PolicyLogger.debug(s"Policy generation JS eveluation of directive parameter timeout: ${jsTimeout} s (change with REST API settings parameter 'rudder_generation_jsTimeout')")
+
+    implicit val parallelism = Parallelism(maxParallelism, Scheduler.io(executionModel = ExecutionModel.AlwaysAsyncExecution), TaskSemaphore(maxParallelism = maxParallelism))
+
     val result = for {
       // trigger a dynamic group update
-      computeGroups        <- triggerNodeGroupUpdate()
+      _                    <- if(computeDynGroupsEnabled) {
+                                triggerNodeGroupUpdate()
+                              } else {
+                                PolicyLogger.warn(s"Computing dynamic groups disable by REST API settings 'rudder_generation_compute_dyngroups'")
```

Comment:
Computing dynamic groups at the beginning of the policy generation disabled by REST API settings blabla